coreneuron/nrniv/corenrn_parameters.cpp

```diff
@@ -0,0 +1,174 @@
+    sub_parallel -> add_flag("--skip-mpi-finalize", this->skip_mpi_finalize, "Do not call mpi finalize.");
+
+    auto sub_spike = app.add_subcommand("spike", "Spike exchange options.");
+    sub_spike -> add_option("--ms-phases", this->ms_phases, "Number of multisend phases, 1 or 2.", true)
+        ->check(CLI::Range(1, 2));
+    sub_spike -> add_option("--ms-subintervals", this->ms_subint, "Number of multisend subintervals, 1 or 2.", true)
+        ->check(CLI::Range(1, 2));
+    sub_spike -> add_flag("--multisend", this->multisend, "Use Multisend spike exchange instead of Allgather.");
+    sub_spike -> add_option("--spkcompress", this->spkcompress, "Spike compression. Up to ARG are exchanged during MPI_Allgather.", true)
+        ->check(CLI::Range(0, 100000));
+    sub_spike->add_flag("--binqueue", this->binqueue, "Use bin queue." );
+
+    auto sub_config = app.add_subcommand("config", "Config options.");
+    sub_config -> add_option("-b, --spikebuf", this->spikebuf, "Spike buffer size.", true)
+        ->check(CLI::Range(0, 2000000000));
+    sub_config -> add_option("-g, --prcellgid", this->prcellgid, "Output prcellstate information for the gid NUMBER.")
+        ->check(CLI::Range(-1, 2000000000));
+    sub_config -> add_option("-k, --forwardskip", this->forwardskip, "Forwardskip to TIME")
+        ->check(CLI::Range(0., 1e9));
+    sub_config -> add_option("-l, --celsius", this->celsius, "Temperature in degC. The default value is set in defaults.dat or else is 34.0.", true)
+        ->check(CLI::Range(-1000., 1000.));
+    sub_config -> add_option("-x, --extracon", this->extracon, "Number of extra random connections in each thread to other duplicate models.")
+        ->check(CLI::Range(0, 10000000));
+    sub_config -> add_option("-z, --multiple", this->multiple, "Model duplication factor. Model size is normal size * multiple")
+        ->check(CLI::Range(1, 10000000));
+    sub_config -> add_option("--mindelay", this->mindelay, "Maximum integration interval (likely reduced by minimum NetCon delay).", true)
+        ->check(CLI::Range(0., 1e9));
+    sub_config -> add_option("--report-buffer-size", this->report_buff_size, "Size in MB of the report buffer.")
+        ->check(CLI::Range(1, 128));
+
+    auto sub_output = app.add_subcommand("output", "Output configuration.");
+    sub_output -> add_option("-i, --dt_io", this->dt_io, "Dt of I/O.", true)
+        ->check(CLI::Range(-1000., 1e9));
+    sub_output -> add_option("-o, --outpath", this->outpath, "Path to place output data files.", true)
+        ->check(CLI::ExistingPath);
+    sub_output -> add_option("--checkpoint", this->checkpointpath, "Enable checkpoint and specify directory to store related files.")
+        ->check(CLI::ExistingDirectory);
+};
+
+void corenrn_parameters::parse (int argc, char** argv) {
+
+    try {
+        app.parse(argc, argv);
+    } catch (const CLI::ParseError &e) {
```

Comment:
Would calling with `-mpi` cause a ParseError? If so, maybe the catch should print out a message to tell the user what exactly is the issue (that we moved from single-dash to double-dash long-parameter syntax).

Reply:
See above. It would be a great feature. I'll try to add it.

Reply:
see if `CLI::ParseError` can provide the necessary information.

Reply:
@ohm314 : `CLI::ParseError` gives access to name of the error and error message itself:

```
The following argument was not expected: -mpi
```

we can parse that error message. But, we can instead go through argv and check if old argument like `-mpi` is used.

Reply:
I dug into the CLI11 exception code and indeed there seems to be no nice way to simply access those bad arguments. Then I guess for now we can rely on CLI11 returning its error message.

Reply:
I would maybe add additionally add something like:
```
cerr << "Single-dash arguments such as -mpi are depricated, please check ./coreneuron --help for more information." << std::endl
```